### PR TITLE
SmtEngine::getAssignment now returns a vector of assignments.

### PR DESCRIPTION
**Pull request by aniemetz**
This is in preparation of the new API.

### CODE REVIEW

File: src/smt/smt_engine.h [hunk 1]
```diff
@@ -598,7 +598,8 @@ class CVC4_PUBLIC SmtEngine {
    * INVALID query).  Only permitted if the SmtEngine is set to
    * operate interactively and produce-assignments is on.
    */
-  CVC4::SExpr getAssignment();
+  //CVC4::SExpr getAssignment();
```
Comment (4tXJ7f) [inline src/smt/smt_engine.h]: This line can be removed :)

File: src/smt/command.cpp [hunk 2]
```diff
@@ -1321,7 +1321,23 @@ void GetAssignmentCommand::invoke(SmtEngine* smtEngine)
...
   {
-    d_result = smtEngine->getAssignment();
+    std::vector<std::pair<Expr,Expr>> assignments  = smtEngine->getAssignment();
+    vector<SExpr> sexprs;
+    for (const auto& p : assignments)
+    {
+      vector<SExpr> v;
+      if (p.first.getKind() == kind::APPLY)
+      {
+        v.push_back(SExpr(SExpr::Keyword(p.first.getOperator().toString())));
```
Comment (4tXJ7f) [inline src/smt/command.cpp]: Could use `std::vector::emplace_back()` here.

File: src/smt/command.cpp [hunk 3]
```diff
@@ -1321,7 +1321,23 @@ void GetAssignmentCommand::invoke(SmtEngine* smtEngine)
...
+      }
+      else
+      {
+        v.push_back(SExpr(SExpr::Keyword(p.first.toString())));
```
Comment (4tXJ7f) [inline src/smt/command.cpp]: Ditto

File: src/smt/command.cpp [hunk 4]
```diff
@@ -1321,7 +1321,23 @@ void GetAssignmentCommand::invoke(SmtEngine* smtEngine)
...
+      }
+      v.push_back(SExpr(SExpr::Keyword(p.second.toString())));
```
Comment (4tXJ7f) [inline src/smt/command.cpp]: Ditto

File: src/smt/command.cpp [hunk 5]
```diff
@@ -1321,7 +1321,23 @@ void GetAssignmentCommand::invoke(SmtEngine* smtEngine)
...
+      sexprs.push_back(SExpr(v));
```
Comment (4tXJ7f) [inline src/smt/command.cpp]: Ditto

File: src/smt/smt_engine.cpp [hunk 6]
```diff
@@ -5170,50 +5170,45 @@ CVC4::SExpr SmtEngine::getAssignment() {
...
-  vector<SExpr> sexprs;
-  TypeNode boolType = d_nodeManager->booleanType();
-  TheoryModel* m = d_theoryEngine->getModel();
-  for(AssignmentSet::key_iterator i = d_assignments->key_begin(),
-        iend = d_assignments->key_end();
-      i != iend;
-      ++i) {
-    Assert((*i).getType() == boolType);
+  vector<pair<Expr,Expr>> res;
+  if (d_assignments != NULL)
```
Comment (4tXJ7f) [inline src/smt/smt_engine.cpp]: `nullptr`?

File: src/smt/smt_engine.cpp [hunk 7]
```diff
@@ -5170,50 +5170,45 @@ CVC4::SExpr SmtEngine::getAssignment() {
...
-    vector<SExpr> v;
-    if((*i).getKind() == kind::APPLY) {
-      Assert((*i).getNumChildren() == 0);
-      v.push_back(SExpr(SExpr::Keyword((*i).getOperator().toString())));
-    } else {
-      Assert((*i).isVar());
-      v.push_back(SExpr(SExpr::Keyword((*i).toString())));
+      Assert(as.getKind() == kind::APPLY || as.isVar());
+      Assert(as.getKind() != kind::APPLY || as.getNumChildren() == 0);
+      res.push_back(make_pair(as.toExpr(), resultNode.toExpr()));
```
Comment (4tXJ7f) [inline src/smt/smt_engine.cpp]: Could use `std::vector::emplace_back()` here.

File: src/smt/command.cpp (hunk 2 above)
Comment (aniemetz) [reply src/smt/command.cpp]: Done

File: src/smt/command.cpp (hunk 4 above)
Comment (aniemetz) [reply src/smt/command.cpp]: Done

File: src/smt/command.cpp (hunk 5 above)
Comment (aniemetz) [reply src/smt/command.cpp]: Done

File: src/smt/smt_engine.cpp (hunk 6 above)
Comment (aniemetz) [reply src/smt/smt_engine.cpp]: Done

File: src/smt/smt_engine.cpp (hunk 7 above)
Comment (aniemetz) [reply src/smt/smt_engine.cpp]: Done

**Pull request merged**